### PR TITLE
WRP-10559: Update dependencies to new major version

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Upgrade dependencies to major versions. 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Performed upgrades and code adjustments, where necessary, for:
**"node-fetch": "^3.3.1"** - latest version, required just changing the 'node-fetch' import to a dynamic import
**"react-router-dom": "^6.9.0"** - latest version, required some code refactoring to use '<Routes>', 'element' and 'useLocation' hook
**"jest": "^28.1.3"** - upgraded to latest version that still supports node v12 (could not upgrade to v29.5.0: https://github.com/facebook/jest/blob/main/CHANGELOG.md, PR: https://github.com/facebook/jest/pull/13033)
**"jest-environment-jsdom": "^28.1.3"** - upgraded to latest version that still supports node v12 (needed to be added separately as starting 28.0.0 they stop shipping 'jest-environment-jsdom' by default, PR: https://github.com/facebook/jest/pull/12354)
**"wait-on": "^7.0.1"** -  - latest version, no code adjustment necessary

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
"puppeteer" and "puppeteer-core" packages could not be upgraded from "^18.2.1" to "^19.7.4". We tried, added the executablePath option like here: https://stackoverflow.com/questions/74251875/puppeteer-error-an-executablepath-or-channel-must-be-specified-for-puppete. But, we got the following error when running the tests "ws does not work in the browser. Browser clients must use the native WebSocket object".

### Links
[//]: # (Related issues, references)
WRP-10559

### Comments

### CODE REVIEW

File: src/views/ScrollerMultipleChildren.js [hunk 1]
```diff
@@ -9,13 +10,14 @@ const types = {
...
-const ScrollerMultipleChildren = ({location}) => {
+const ScrollerMultipleChildren = () => {
+	const location = useLocation();
 	const search = qs.parse(location.search, {ignoreQueryPrefix: true});
 	const type = search.type;
 	const Scroller = types[type] || ScrollerJS;
 
 	return (
-		<Scroller id="Scroller" animate>
+		<Scroller id="Scroller">
```
Comment (stanca-pop-lgp) [inline src/views/ScrollerMultipleChildren.js:20]: *Note: 'animate' prop was deleted as it caused a console error: "'Received "true" for a non-boolean attribute". 'animate' is not a prop of Scroller, it's a parameter for the 'scrollTo' function, so we believe it is not needed here.

**Review by alexandrumorariu — APPROVED** [review]
LGTM

File: performance/TestResults.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require('dotenv').config();
 const  fs = require('fs');
 const path = require('path');
-const fetch = require('node-fetch');
+const fetchAPI = (...args) => import('node-fetch').then(({default: fetchData}) => fetchData(...args));
```
Comment (hong6316) [inline performance/TestResults.js:4]: Could you try this?
line 4: const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
line 20: (revert the change)
(Refenece: https://github.com/node-fetch/node-fetch/blob/HEAD/docs/v3-UPGRADE-GUIDE.md#converted-to-es-module )

Or there's any reason to rename `fetchAPI`?

**Comment by hong6316** [comment]
There's another plan or ticket to update "puppeteer" and "puppeteer-core"?

File: performance/TestResults.js (hunk 2 above)
Comment (stanca-pop-lgp) [reply performance/TestResults.js:4]: Yes, we get a lint error:
error    'fetch' is already a global variable  no-shadow

**Comment by stanca-pop-lgp** [comment]
> There's another plan or ticket to update "puppeteer" and "puppeteer-core"?

We will add it on our future tickets list. For now, we were not able to upgrade those packages.
